src/nvim/statusline.c

```diff
@@ -79,6 +79,10 @@ void win_redr_status(win_T *wp)
   busy = true;
 
   wp->w_redr_status = false;
+  if (wp->w_floating && is_stl_global) {
+    busy = false;
```

Comment:
why not add this check to the condition on line 76? then wouldn't need to set busy=false here, and logic lives in one place.

oh, is the `wp->w_redr_status = false` assignment important?